packages/downloads/src/request-download-metadata.ts

```diff
@@ -24,28 +24,31 @@ export interface IDownloadMetadataRequestParams {
 
 export interface IDownloadMetadataResults {
   /* Identifier for the download */
-  downloadId: string,
+  downloadId: string;
 
-  /* ready, not_ready, creating, updating, failed*/
-  status: string,
+  /* ready, not_ready, creating, updating, failed */
+  status: string;
```

Comment:
why isn't this an enum?

Reply:
or a `type` like `'ready' | 'not_ready' | 'creating' | 'updating' | 'failed'`?

That's what we use in other places for this sort of thing.

@mikeringrose I still don't have good instincts on when to use one of the many ways (`type`, `enum`, `const enum`, etc) TS offers for this kind of situation. Worse, I'm sort of mentally blocked on it after having been [burned by making the wrong choice over in rest-js](https://github.com/Esri/arcgis-rest-js/pull/681). If you have experience/insight, I think it would be great to document some guidelines here.

Reply:
@mikeringrose - I'm using a `DownloadStatus` type now.